pkg/render/intrusion_detection.go

```diff
@@ -191,6 +195,10 @@ func (c *intrusionDetectionComponent) intrusionDetectionJobContainer() v1.Contai
 				Name:  "CLUSTER_NAME",
 				Value: c.esClusterConfig.ClusterName(),
 			},
+			{
+				Name:  "ELASTIC_LICENSE_TYPE",
+				Value: string(c.elasticLicenseType),
```

Comment:
Does it matter how Intrusion treats an empty ELASTIC_LICENSE_TYPE?
Is there any problem if initially this uses the wrong License?
It seems likely that this can be ran with the wrong value since it is a job. Since the controller will launch this if the license does not exist, I'm concerned on the impact if it is wrong.
What I'm kind of getting at is, should the controller require the license config?

Reply:
Nevermind, I'm wrong. I see the controller bails if the license type is not found.

Reply:
If ELASTIC_LICENSE_TYPE is set to basic, we skip installing ml jobs in es-job-installer. If it is empty or has a unknown value, it will continue as usual, that is install ml jobs and apply kibana dashboards. So it is basically the same behaviour as before, except we don't install ml jobs if license is basic.

>  Is there any problem if initially this uses the wrong License?

Say if we switch from basic to enterprise, since we are watching `elastic-licensing` configmap, this job will get restarted and ml jobs will get installed, shouldn't be any problem.
For other way around, enterprise to basic, as xpack is not available anymore ml jobs will not run.

I haven't changed anything in the intrusion-detection-controller, it will continue watching for ml jobs and create reports/alerts if job is available, else log error. It does seem to be flood with error, I'll add logic in operator to set `DISABLE_ANOMALY=yes` if license type is basic to avoid all the errors.